doc/negative-http2-interop-test-descriptions.md

```diff
@@ -0,0 +1,193 @@
+
+Client
+------
+
+Clients implement test cases that test certain functionally. Each client is
+provided the test case it is expected to run as a command-line parameter. Names
+should be lowercase and without spaces.
+
+Clients should accept these arguments:
+* --server_host=HOSTNAME
+    * The server host to connect to. For example, "localhost" or "127.0.0.1"
+* --server_port=PORT
+    * The server port to connect to. For example, "8080"
+* --test_case=TESTCASE
+    * The name of the test case to execute. For example, "goaway"
+
+Note
+-----
+
+Note that the server and client must be invoked with the same test case or else
+the test will be meaningless. For convenience, we provide a shell script wrapper
+that invokes both server and client at the same time, with the same test_case.
+This is the preferred way to run these tests.
+
+## Test Cases
+
+### goaway
+
+This test verifies that the client correctly responds to a goaway sent by the
+server. The client should handle the goaway by switching to a new stream without
+the user application having to do a thing.
+
+Client Procedure:
+ 1. Client sends two UnaryCall requests with:
+ 
+     ```
+    {
+      response_size: 1024
+      payload:{
+        body: 1024 bytes of zeros
+      }
+    }
+    ```
+
+Client asserts:
+* call was successful.
+* response payload body is 1024 bytes in size.
+
+Server Procedure:
+  1. Server sends a GOAWAY after receiving the first UnaryCall.
+
+Server asserts:
+* The second UnaryCall has a different stream_id than the first one.
+
+### rst_after_header
+
+This test verifies that the client fails "correctly" when the server sends a
+RST_STREAM immediately after sending headers to the client.
+
+Procedure:
+ 1. Client sends UnaryCall with:
+ 
+  ```
+  {
+    response_size: 1024
+    payload:{
+      body: 1024 bytes of zeros
+    }
+  }
+  ```
+
+Client asserts:
+* Call was not successful
+
+Server Procedure:
+  1. Server sends a RST_STREAM with error code 0 after sending headers to the client.
+  
+*At the moment the error code and message returned are not standardized throughout all
+languages. Those checks will be added once all client languages behave the same way.*
+
+### rst_during_data
+
+This test verifies that the client fails "correctly" when the server sends a
+RST_STREAM halfway through sending data to the client.
+
+Procedure:
+ 1. Client sends UnaryCall with:
+ 
+  ```
+  {
+    response_size: 1024
+    payload:{
+      body: 1024 bytes of zeros
+    }
+  }
+  ```
+
+Client asserts:
+* Call was not successful
+
+Server Procedure:
+  1. Server sends a RST_STREAM with error code 0 after sending half of 
+     the requested data to the client.
+
+### rst_after_data
+
+This test verifies that the client fails "correctly" when the server sends a
+RST_STREAM after sending all of the data to the client.
+
+Procedure:
+ 1. Client sends UnaryCall with:
+  ```
+  {
+    response_size: 1024
+    payload:{
+      body: 1024 bytes of zeros
+    }
+  }
```

Comment:
This code block is all showing on one line in Github's preview: https://github.com/ncteisen/grpc/blob/fcb2d1470664419a79692610f9aa60587f0aedc1/doc/negative-http2-interop-test-descriptions.md

It's fine in other markdown editor previews, not sure if it needs a newline before the start of the codeblock or something else is happening?